src/amqp_management.c

```diff
@@ -54,8 +54,8 @@ typedef struct AMQP_MANAGEMENT_INSTANCE_TAG
     AMQP_MANAGEMENT_STATE amqp_management_state;
     char* status_code_key_name;
     char* status_description_key_name;
-    int sender_connected : 1;
-    int receiver_connected : 1;
+    uint8_t sender_connected : 1;
```

Comment:
>uint8_t [](start = 4, length = 7)

C99 supports both signed and unsigned bitfields. Is the compiler you are using complaining about int?

Basically I am trying to figure out whether this is a needed change to take in or not.

The reason is that some compilers we had to deal with did not like unsigned bitfields, so we switched at some point to int only.

Thus I'd like to about going back and forth between signed and unsigned.

Reply:
For some reason I see that connection has unsigned int and that works fine with the compilers we had issues on. Could you just make this unisgned int also to have them all in sync?
Thanks!

---
In reply to: [188157348](https://github.com/Azure/azure-uamqp-c/pull/227#discussion_r188157348) [](ancestors = 188157348)

Reply:
"../../src/amqp_management.c", line 55: warning: signed bit field of length 1

is the specific warning. I guess you could have a signed bit field of length 2 that could contain 1,0,-1,-2 if I remember my 2s complement math correctly.

The only reason I chose uint8_t over unsigned int was that uint64_t was already being used in this same structure and I didn't know if you were passing the structure between different machine where the size of an int varying by compiler might pose an issue. 

At the end of the day all these changes are offered as feedback that we uncovered while porting Azure IoT SDK to the TI CC3220 so I'm not heavily invested in whether you take them or not, you won't hurt my feelings. I just wanted you to be able to realize value from the work we did.

Reply:
This should be perfectly fine as unisgned int or uint8_t. SO I'll leave it as it is in your PR and if I will decide to make them consistent across modules later that is a different story. Cheers!

---
In reply to: [188477250](https://github.com/Azure/azure-uamqp-c/pull/227#discussion_r188477250) [](ancestors = 188477250)